### PR TITLE
Display FTL version & version.sh rewrite

### PR DESCRIPTION
**Pull request by WaLLy3K**
**By submitting this pull request, I confirm the following:**

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:**
8

---

While testing to make sure `pihole -v` would output `pihole-FTL version`, I noticed some options didn't work how I expected them to. For example, if I use `pihole -v -p`, I would expect to see the version output of Pi-hole Core. Instead, I'm informed that it's an invalid option.

I've had the following things in mind while rewriting this:
  * I'm operating under the assumption that FTL is only installed if the Admin Console is (Line 113 exit 0)
  * I have modified the help text to only output with `pihole -v --help`
  * I have modified all output to be more similar to the output style of `grep` and `curl` (Ditching ":::")

Testing output:
```
w3k@MCT:~$ pihole -v
  Pi-hole version is v3.0.1-14-ga928cd3 (Latest: v3.0.1)
  Admin Console version is v3.0-9-g3760482 (Latest: v3.0.1)
  FTL version is v2.6.2 (Latest: v2.6.2)
w3k@MCT:~$ pihole -v -c
  Current Pi-hole version is v3.0.1-14-ga928cd3
  Current Admin Console version is v3.0-9-g3760482
  Current FTL version is v2.6.2
w3k@MCT:~$ pihole -v -l
  Latest Pi-hole version is v3.0.1
  Latest Admin Console version is v3.0.1
  Latest FTL version is v2.6.2
w3k@MCT:~$ pihole -v -p --hash
  Current Pi-hole hash is a928cd3
w3k@MCT:~$ pihole -v -a --hash
  Current Admin Console hash is 3760482
w3k@MCT:~$ pihole -v --help
Usage: pihole -v [REPO | OPTION] [OPTION]
Show Pi-hole, Web Admin & FTL versions
  <Shows all Repositories and Options>
w3k@MCT:~$ pihole -v -foo
  Invalid Option!
```

### CODE REVIEW

**Comment by Mcat12** [comment]
I think these commands should be working, based on the help text:
```
pi@raspberrypi:/etc/.pihole $ pihole -v -a -h
  Invalid Option! Try 'pihole -v --help' for more information.
pi@raspberrypi:/etc/.pihole $ pihole -v -p -h
  Invalid Option! Try 'pihole -v --help' for more information.
pi@raspberrypi:/etc/.pihole $ pihole -v -p --hash
  Current Pi-hole hash is 03201e2
```
And this one has invalid input but continues as if it was just `-v`, so I'm not sure if that's intended or if it should tell the user there's invalid input:
```
pi@raspberrypi:/etc/.pihole $ pihole -v -c -h
  Current Pi-hole version is v3.0-35-g03201e2
  Current Admin Console version is v3.0-10-g9cf967e
  Current FTL version is v2.6.2
```

**Comment by WaLLy3K** [comment]
You're absolutely right, thanks for spotting it! Latest commit fixes that.

```
w3k@MCT:~$ pihole -v -a -h
  Current Admin Console hash is 3760482
w3k@MCT:~$ pihole -v -p -h
  Current Pi-hole hash is a928cd3
w3k@MCT:~$ pihole -v -c -h
  Invalid Option! Try 'pihole -v --help' for more information.
```

File: advanced/Scripts/version.sh [hunk 1]
```diff
@@ -80,17 +81,24 @@ coreOutput() {
...
     str="Current Pi-hole hash is $hash"
   else
-    echo "  Invalid Option! Try 'pihole -v --help' for more information."
-    exit 1
+	error="true"
+  fi
+
+  if [ "$error" = "true" ]; then
+	echo "  Invalid Option! Try 'pihole -v --help' for more information."
+	exit 1
```
Comment (Mcat12) [inline advanced/Scripts/version.sh]: Looks like there's no indentation right here.

File: advanced/Scripts/version.sh [hunk 2]
```diff
@@ -101,16 +109,22 @@ webOutput() {
...
     str="Current Admin Console hash is $hash"
   else
-    echo "  Invalid Option! Try 'pihole -v --help' for more information."
-    exit 1
+	error="true"
   fi
+
+  if [ "$error" = "true" ]; then
+	echo "  Invalid Option! Try 'pihole -v --help' for more information."
+	exit 1
```
Comment (Mcat12) [inline advanced/Scripts/version.sh]: Indentation here and line 112

File: advanced/Scripts/version.sh [hunk 3]
```diff
@@ -119,16 +133,21 @@ ftlOutput() {
...
     str="Latest FTL version is $latest"
   else
-    echo "  Invalid Option! Try 'pihole -v --help' for more information."
-    exit 1
+	error="true"
   fi
+
+  if [ "$error" = "true" ]; then
+	echo "  Invalid Option! Try 'pihole -v --help' for more information."
+	exit 1
```
Comment (Mcat12) [inline advanced/Scripts/version.sh]: Indentation here and line 136

**Comment by Mcat12** [comment]
...what?
image

image

**Comment by Mcat12** [comment]
It looks like Github has mixed feelings about tabs...

**Comment by WaLLy3K** [comment]
The latest commit fixes the output error of (`pihole -v -h`), and cleans up the three output functions into one neat one.

The only known issue is the output of (`pihole -v -f -h`) as mentioned below:
```
pihole -v --help
Usage: pihole -v [REPO | OPTION] [OPTION]
Show Pi-hole, Web Admin & FTL versions
  <Shows more information>
  
pihole -v
  Pi-hole version is v3.0.1-14-ga928cd3 (Latest: v3.0.1)
  AdminLTE version is v3.0-9-g3760482 (Latest: v3.0.1)
  FTL version is v2.6.2 (Latest: v2.6.2)
pihole -v -c
  Current Pi-hole version is v3.0.1-14-ga928cd3
  Current AdminLTE version is v3.0-9-g3760482
  Current FTL version is v2.6.2
pihole -v -l
  Latest Pi-hole version is v3.0.1
  Latest AdminLTE version is v3.0.1
  Latest FTL version is v2.6.2
pihole -v -h
  Current Pi-hole hash is a928cd3
  Current AdminLTE hash is 3760482
pihole -v -foo
  Invalid Option! Try 'pihole -v --help' for more information.

pihole -v -a
  AdminLTE version is v3.0-9-g3760482 (Latest: v3.0.1)
pihole -v -a -c
  Current AdminLTE version is v3.0-9-g3760482
pihole -v -a -l
  Latest AdminLTE version is v3.0.1
pihole -v -a -h
  Current AdminLTE hash is 3760482
pihole -v -a -foo
  Invalid Option! Try 'pihole -v --help' for more information.
  
pihole -v -p
  Pi-hole version is v3.0.1-14-ga928cd3 (Latest: v3.0.1)
pihole -v -p -c
  Current Pi-hole version is v3.0.1-14-ga928cd3
pihole -v -p -l
  Latest Pi-hole version is v3.0.1
pihole -v -p -h
  Current Pi-hole hash is a928cd3
pihole -v -p -foo
  Invalid Option! Try 'pihole -v --help' for more information.

pihole -v -f
  FTL version is v2.6.2 (Latest: v2.6.2)
pihole -v -f -c
  Current FTL version is v2.6.2
pihole -v -f -l
  Latest FTL version is v2.6.2
pihole -v -f -h
  <No error, but no output. I can't think of a fix for this right now>
pihole -v -f -foo
  Invalid Option! Try 'pihole -v --help' for more information.
```

**Comment by dschaper** [comment]
Would it be a good idea to include the hash of the latest version as well? To see if the users have made any commits to their branches?

**Comment by WaLLy3K** [comment]
If there's an API that allows us to retrieve the latest hash for `master` as well as all other branch, then that's definitely something I'll implement.

Until then, I don't think there's anything else of value I can add to this PR. :smile:

**Comment by dschaper** [comment]
git ls-remote --heads https://github.com/pi-hole/pi-hole works, would need some awk and bracework. Can specify branch to drill down.

**Comment by dschaper** [comment]
Remote hashes probably fall under gold plating and would be better as a separate PR. Will merge this as is.

**Pull request merged**